### ISSUE TITLE
EPIC: auto-generate artifacthub metadata and consume it on policies

### ISSUE THREAD

**Issue by viccuad** [kubewarden/kubewarden-controller#389]
We have defined the usage of `artifacthub-pkg.yml` fields in UI, and how to derive them from `metadata.yml`.
We have hand-crafted `artifacthub-pkg.yml` for the policies to unblock the UI work.

Still, we are missing automating the generation of `artifacthub-pkg.yml`:
- [x] RFC for implementation approach and alternatives:  https://github.com/kubewarden/rfc/pull/15
- [x] https://github.com/kubewarden/kwctl/issues/418
- [x] kubewarden/github-actions#48
- [x] Update policy templates if needed, so future community policies are easier to publish on artifacthub
- [x] https://github.com/kubewarden/docs/pull/173
- [x] https://github.com/kubewarden/kubewarden-controller/issues/325
- [ ] #418

**Comment by flavio** [comment]
> Update GHA workflows to use the new kwctl command, and, either generate the new artifacthub-pkg.yml, or make sure it has automatically been updated and committed, prior to release

We cannot generate the artifacthub-pkg as part of our pipeline. The workflow of the policy author must be something like that:

1. Generate `artifacthub-pkg.yml` via the scaffold command
2. Commit the file into git
3. Create a new `v<something` tag

IMHO the `artifacthub-pkg.yml` contents should follow the same versioning of the policy, meaning that checking out a specific tag of the policy repository will show what the metadata of ArtifactHUB looked like.

Moreover, when I did the last round of updates of policies, it turned out that ArtifactHUB scanner takes into account the version of the policy mentioned inside of the `artifacthub-pkg.yml` file in order to update its database.

Long story short:
* `artifacthub-pkg.yml` says the policy version is `0.1.0`
* -> AH scan updates the database
* Ouch, we have a typo inside of the `artifacthub-pkg.yml` file, let's fix it and update the `createdAt` field
* -> AH scan the repo, but doesn't change the database
* Ouch, the metadata on AH is stale. Let's bump the `version` to be `0.1.1` inside of `artifacthub-pkg.yml`
* -> AH scan updates the database

Sorry for the long explanation, that's to say... we need to bump the `version` attribute inside of the `artifacthub-pkg.yml` file to have the AH database updated. Hence we have to tie this operation with a git tag.

**Comment by flavio** [comment]
> Re-release policies

In theory we don't have to do it. The data inside of `artifacthub-pkg.yml` should already be correct. We can slowly regenerate that with future releases of the policies

**Comment by flavio** [comment]
> Update policy templates if needed, so future community policies are easier to publish on artifacthub

Yes, to be more specific. I would love to have new `Makefile` targets that: generates the `artifacthub-pkg.yml` file using the `scaffold` command

**Comment by flavio** [comment]
>  Update GHA workflows to use the new kwctl command, and, either generate the new artifacthub-pkg.yml, or better, make sure it has automatically been generated, as we do with metadata.yml, prior to release

More about that, I think we should have a check that makes sure that, when doing a policy release, the TAG pushed matches with the version of the package inside of `Cargo.toml` (if that applies) and the value inside of `artifacthub-pkg.yml`.

**Comment by viccuad** [comment]
It seems that [artifacthub provides a cli, `ah`](https://artifacthub.io/docs/topics/cli/). Maybe we can edit the templates so, if uncommenting "publish to artifacthub", the correct GHA workflow will be run, the artifacthub-pkg checked, the version compared, etc, so policy authors don't need a flowchart just to build a policy.

**Comment by flavio** [comment]
Moving to blocked, the community policies cannot be automatically published on ArtifactHub because of this issue: https://github.com/kubewarden/github-actions/issues/70

**Comment by flavio** [comment]
Closing, everything has been done. Now we're waiting for the community to update their policies
